tests/test_guru.py

```diff
@@ -1,61 +1,63 @@
-            if not _empty(i):
-                return False
-    elif isinstance(obj, list) or isinstance(obj, tuple):
-        for i in obj:
-            if not _empty(i):
-                return False
-    else:
-        return True
-    return True
-
-
-def test_without_warnings():
-    guru = Guru(print_hints=False)
-    X, y = load_breast_cancer(True)
-    result = guru.check_data(XYCDataset(X, y, None))
-    assert _empty(result)
-
-
-def test_warnings():
-    guru = Guru(print_hints=False)
-    result = guru.check_data(_TOY_DATASET, category_qoute=0.5)
-    assert result == ({'not numeric': [0], 'not variable': [3, 4]},
-                      [5], {'too rare': [1], 'too common': [0]})
-
-
-def test_sparse_arguments():
-    guru = Guru(print_hints=False)
-
-    params = [{'check_sparsity': False},
-              {'sparse_qoute': 0.3}]
-    answers = [(None, None, None),
-               (None, [5, 6], None)]
-
-    for param, answer in zip(params, answers):
-        assert answer == guru.check_data(_TOY_DATASET,
-                                         check_categorial_features=False,
-                                         check_class_disbalance=False,
-                                         **param)
+
+import numpy as np
+
+
+_TOY_X = [['joke', 1231, 0.12312, 1, True, 0, 0],
+          ['lol', 123, 1231.1231, 1, False, 0, 0],
+          ['not joke', 1313, 12.133, 1, False, 0, 0],
+          ['joke', 12312, 0.24183, 0, True, 1, 0],
+          ['silly joke', 321, 0.12, 1, False, 0, 0],
+          ['silly joke', 32, 0.2, 1, False, 0, 0], ['silly joke', 31, 0.1, 1, True, 0, 1.1],
+          ['silly joke', 21, 0.123, 1, True, 0, 4.6],
+          ['silly joke', 21, 0.123, 1, True, 0, 0],
+          ['joke', 1, 0.23, 1, False, 0, 0],
+          ['silly joke', 1, 0.124, 1, True, 0, -2.73]]
+_TOY_Y = [0]*8 + [1] + [3]*2
+_BREAST_X, _BREAST_Y = load_breast_cancer(True)
+
+
+def test_sparse():
+    guru_params = [{}, {}, {'sparse_qoute': 0.5}]
+    Xs = [_BREAST_X] + [_TOY_X] * 2
+    answers = [[], [5], [5, 6]]
+
+    _test(lambda guru, data: guru.check_sparse(data), guru_params, Xs, answers)
+
+
+def test_categorial():
+    guru_params = [{}, {}, {'category_qoute': 0.15}]
+    args = [[_BREAST_X]] + [[_TOY_X]] * 2
+    answers = [{}, {Guru._NOT_NUMERIC_KEY: [0], Guru._NOT_VARIABLE_KEY: [3, 4]},
+               {Guru._NOT_NUMERIC_KEY: [0], Guru._NOT_VARIABLE_KEY: [3]}]
+    _test(lambda guru, args: guru.check_categorial(*args), guru_params, args, answers)
+
+
+def test_class_disbalance():
+    guru_params = [{}, {}, {'class_disbalance_qoute': 0.4},
+                   {'class_disbalance_qoute': 0.8}]
+    args = [[_BREAST_Y]] + [[_TOY_Y]] * 2
+    answers = [{},
+               {Guru._TOO_COMMON_KEY: [0], Guru._TOO_RARE_KEY: [1]},
+               {Guru._TOO_RARE_KEY: [1]},
+               {Guru._TOO_COMMON_KEY: [0], Guru._TOO_RARE_KEY: [1, 3]}]
+    _test(lambda guru, args: guru.check_class_disbalance(*args), guru_params, args, answers)
```

Comment:
why do you need lambda here? can't you just pass `guru.check_class_disbalance`?

Reply:
I need different Guru objects to test different parameters of Guru.__init__

Reply:
why can't you init Guru once before _test?